skglm/solvers/group_prox_newton.py

```diff
@@ -0,0 +1,307 @@
+import numpy as np
+from numba import njit
+from numpy.linalg import norm
+from skglm.solvers.base import BaseSolver
+from skglm.utils import check_group_compatible
+
+EPS_TOL = 0.3
+MAX_CD_ITER = 20
+MAX_BACKTRACK_ITER = 20
+
+
+class GroupProxNewton(BaseSolver):
```

Comment:
thanks Badr, it looks really cool. Do you think there is hope to reduce code redundancy with the vanilla prox newton solver ?

Reply:
It's possible if we add ``fit_intercept`` to the solver.

Reply:
go ahead if it reduces the code duplication